k8s.tf

```diff
@@ -125,12 +133,53 @@ resource "azurerm_kubernetes_cluster_node_pool" "execution-nodes" {
   tags = var.tags
 
   lifecycle {
-      ignore_changes = [
-          availability_zones,
-          enable_host_encryption,
-          enable_node_public_ip,
-          vnet_subnet_id,
-          node_count
-      ]
+    ignore_changes = [
+      availability_zones,
+      enable_host_encryption,
+      enable_node_public_ip,
+      vnet_subnet_id,
+      node_count
+    ]
   }
 }
+
+resource "azurerm_kubernetes_cluster_node_pool" "gpu-execution-nodes" {
+  count                 = var.gpuNodePool ? 1 : 0
+  name                  = "gpuexecnodes"
+  mode                  = "User"
+  orchestrator_version  = var.kubernetesVersion
+  os_disk_size_gb       = 128
+  kubernetes_cluster_id = azurerm_kubernetes_cluster.aks.id
+  min_count             = var.gpuNodeCountMin
+  max_count             = var.gpuNodeCountMax
+  node_count            = var.gpuNodeCountMin
+  vm_size               = var.gpuNodeSize
+  max_pods              = 50
+  enable_auto_scaling   = true
+  vnet_subnet_id        = azurerm_subnet.gpu-nodes-subnet.id
+
+  node_labels = {
+    "purpose" = "execution" # Todo: clarify if Aurelion exec pods expect specific node labels
```

Comment:
From my point of view the "purpose" must not be "execution" but "gpu". For the taints "purpose=gpu" is also used.

Reply:
Fixed in 5669c7b18fa3e61302870da1d8caf3aa42de5316